gems/sorbet-runtime/lib/types/types/simple.rb

```diff
@@ -60,7 +62,7 @@ def self.type_for_module(mod)
             Simple.new(mod)
           end
 
-          mod.instance_variable_set(:@__as_sorbet_type, type) unless mod.frozen?
+          @cache[mod] = type unless mod.frozen?
```

Comment:
We don't need the `frozen?` check here anymore

Reply:
I thought so initially as well, but then the `frozen module` test failure reminded me why we still need it. Internally, `WeakMap` adds a finalizer to the stored object, so that it can clear that map entry when the object is garbage collected. For a frozen object, though, adding a finalizer is not a valid operation, so this still raises.

Reply:
Huh, weird. That could use a comment probably?

Reply:
Sorry for the delay here, just added the requested comment.